test/integration/testConnection.js

```diff
@@ -1131,31 +1131,35 @@ describe('Connection test - connection pool', function ()
     });
 });
 
-describe('Heartbeat test', function ()
+describe('Connection Test - Heartbeat', function ()
 {
-  var connection = snowflake.createConnection(connOption.valid);
+  let connection;
 
-  it('call heartbeat url', function (done)
+  beforeEach(done =>
   {
-    async.series(
-      [
-        function (callback)
-        {
-          connection.connect(function (err, conn)
-          {
-            assert.ok(!err, JSON.stringify(err));
-            callback();
-          });
-        },
-        function (callback)
-        {
-          connection.heartbeat();
-          callback();
-        }
-      ],
-      function ()
-      {
-        done();
-      });
+    connection = snowflake.createConnection(connOption.valid);
+    testUtil.connect(connection, done);
+  });
+
+  after(done =>
+  {
+    testUtil.destroyConnection(connection, done);
   });
-});
\ No newline at end of file
+
+  it('call heartbeat url with default callback', () =>
+  {
+    connection.heartbeat();
+  });
+
+  it('call heartbeat url with callback', done =>
+  {
+    connection.heartbeat(err => {
+      err && done(err) || done()
```

Comment:
Is this fully correct? What if first `done(err)` returns false? Then the second `done` would also be invoked? This is not the same as ternary operator in java I suppose.

Reply:
it's like ternary operator - I can simplify it to make it easier to read